client/jetpack-cloud/sections/partner-portal/license-list/index.tsx

```diff
@@ -130,41 +68,9 @@ export default function LicenseList(): ReactElement {
 				page={ currentPage }
 			/>
 
-			<LicenseListItem header className="license-list__header">
-				<h2>{ translate( 'License state' ) }</h2>
-
-				<SortButton
-					sortField={ LicenseSortField.IssuedAt }
-					currentSortField={ sortField }
-					currentSortDirection={ sortDirection }
-				>
-					{ translate( 'Issued on' ) }
-				</SortButton>
+			{ ! showNoResults && <LicenseListHeader /> }
 
-				{ filter !== LicenseFilter.Revoked && (
-					<SortButton
-						sortField={ LicenseSortField.AttachedAt }
-						currentSortField={ sortField }
-						currentSortDirection={ sortDirection }
-					>
-						{ translate( 'Assigned on' ) }
-					</SortButton>
-				) }
-
-				{ filter === LicenseFilter.Revoked && (
-					<SortButton
-						sortField={ LicenseSortField.RevokedAt }
-						currentSortField={ sortField }
-						currentSortDirection={ sortDirection }
-					>
-						{ translate( 'Revoked on' ) }
-					</SortButton>
-				) }
-
-				<div>{ /* Intentionally empty header. */ }</div>
-
-				<div>{ /* Intentionally empty header. */ }</div>
-			</LicenseListItem>
+			{ showNoResults && <LicenseListEmpty /> }
```

Comment:
I think it would be better if we move LicenseListEmpty to the bottom so when it is visible it does not visually split the filter bar from the results placeholder while transitioning:

https://user-images.githubusercontent.com/22746396/114049403-192ea080-9894-11eb-8cd0-9304b96cee9c.mov

Bonus points:
While the transition is happening you can barely see the no results label change to the current filter as it resolves that from the context (it will become slightly more apparent that something shifts when we have different subtitles for each filter in the future).
I think adding the animation to the component and passing the filter as a prop rather than the context will solve this micro annoyance as well - if it doesn't and it's a more involved fix then feel free to ignore.

Reply:
>I think it would be better if we move LicenseListEmpty to the bottom

Done!

>I think adding the animation to the component and passing the filter as a prop rather than the context will solve this micro annoyance

That doesn't really fix this specific thing, but it makes way more sense to have the filter as a prop coming from the parent rather than the context. So I've implemented this as you suggested.
I added the filter as a key to the transition component and that avoids the label change issue you mentioned above.